test/stablex/stablex_utils.js

```diff
@@ -18,9 +15,11 @@ const { sendTxAndGetReturnValue } = require("../utilities")
  * @returns {}
  */
 const setupGenericStableX = async function(numTokens = 2, maxTokens = 2 ** 16 - 1) {
+  const MockContract = artifacts.require("MockContract")
```

Comment:
nit: revert? I'm not sure what the implications are for inline requiring vs. top-level, but it seems to be more common to do requires at the top if possible.

Reply:
Yes, I will revert that. I don't even remember changing it. 

This change made no sense I agree. Also, imo, doing it in the top is also not a good idea for a utils file. Although that's a refactor totally out of this PR. Doing it on top means you assume the JS file would be executed within a context, where you expect some magical global vars to be injected.

It's kind of fine for truffle scripts, as an entry point. But the utils shouldn't make any assumption, they should get all the parameters they need.

Some utils solve that by receiving the `artifcats` and pulling from there, but I personally believe, usually is better to be specific about the parameters and name the contracts you rely on.

Reply:
Okay @fleupold , I removed that and I broke my script. 

Now I see why I did the change. As I said this relies in a magic globals. I think someone had the same issue when created the `setup_environment` script, and it seems like he opt in to duplicate the logic from this `setupGenericStableX`

This setup function should receive either the artifacts or the 2 contracts and be refactorized across the project. Imo it should be the contracts.

The thing is this falls way out of this PR, as you see, this function is used a little and I don't want to overcomplicate this PR:

![image](https://user-images.githubusercontent.com/2352112/76847685-bdd1e080-6842-11ea-9ef0-cb06e41498d8.png)

I think a review of the dependencies and some refactor would be beneficial in the project. Not sure if this is to ambitious, but ideal if this refactor would be used to do the typescript jump? Probably there's pseudo-automated tools already to migrate a whole project to TS.

Reply:
To be more specific about the error I was getting is:
* Utils file export some functions that are useful in some scripts
* Initially you would think that by requiring the file no code get's executed (no 
* The moment you require the utils file it's going to try to use `artifacts` that won't be available in that context

This is why this "assumptions" specially for utils is not good

Reply:
My point is that since this file is located inside **test**/stablex/ it should only be used by tests and not by external scripts. If they are useful as top-level utils we should move them to src or some other "shareable".

If you feel this is outside the scope of this PR I can think three options:
1. Create a PR that just moves this function outside of its current location to a shareable one, then rebase this PR on top.
2. Copy the code into utils for now and create a follow up PR that merges them
3. Leave it as is and do a follow up PR extracting the method

Reply:
![image](https://user-images.githubusercontent.com/2352112/76858297-69852b80-6857-11ea-8aad-af6ba77f6ecf.png)

But this is reusing the test data, and those function `makeDeposits`, `placeOrders` are relying on the structure used in `test/resources/examples`

I feel it's OK using test data for a test setup in targeted for a local ganache. And since you have the utils for making the trades and deposits I've just used them (Alex gave me the tip)

I also use other tools that are in the test utilities, like `waitForNSeconds`, since it's targeted for ganache setup.

What's would be your concern?

I can move the script to the test folder if u think it's better

Reply:
It's a good point that `waitForNSeconds` is also used in other scripts and comes from the test utilities.
I still think it would be cleaner to lift scripts that are used for both unit tests as well as scripts into a top-level shared utility (instead of cross referencing scripting and unit tests dependencies). But since this is not yet the case it doesn't have to be changed here.

I think the script should definitely be part of the scripts subfolder.

🚢